### ISSUE TITLE
fasttext for one class classification ?

### ISSUE THREAD

**Issue by anujgupta82** [facebookresearch/fastText#269]
Can fasttext be used for one-class classification task ? 

I have training data of only one class. Can I train a fasttext classifier on it to detect outliers ?

**Comment by cpuhrsch** [comment]
Hello @anujgupta82,

By one class do you mean that each datapoint has the same label? What is an outlier in this case?

Thanks,
Christian

**Comment by anujgupta82** [comment]
Hello @cpuhrsch

Yes every data-point in training set has the same label. OCC(One Class Classifiers) learn a decision boundary, such that given a new (unlabeled) data point, it this lies inside the decision boundary or not. If not, its called an outlier. 

http://scikit-learn.org/stable/modules/outlier_detection.html

Thanks
Anuj

**Comment by cpuhrsch** [comment]
Hello @anujgupta82,

Outlier detection is not something fastText supports out of the box or that we have tried in the past. You'll need to be very careful here to get any reasonable results. [One-Class SVM](http://users.cecs.anu.edu.au/~williams/papers/P126.pdf) has a very principled approach to this problem, from what I can tell, by defining a hyperparameter that lets you tune the degree to which a datapoint is an outlier. This is not something we support yet. I'll add this to the list of future applications because it is very interesting, but I can't help you resolve this issue today.

Thanks,
Christian
